Telefonansage/module.php

```diff
@@ -99,17 +113,57 @@ public function StartCallEx(string $PhoneNumber, string $Text)
             echo $this->Translate('The instance is already calling');
             return;
         }
+
+        if (!$this->checkConfiguration()) {
+            return;
+        }
         $id = VoIP_Connect($this->ReadPropertyInteger('VoIPInstanceID'), $PhoneNumber);
 
         $this->SetBuffer('CallID', json_encode($id));
         $this->SetBuffer('Text', $Text);
         $this->SetTimerInterval('CloseConnectionTimer', 1000 * $this->ReadPropertyInteger('WaitForConnection'));
     }
 
-    public function CloseConnection() {
+    public function CloseConnection()
+    {
+        if (!$this->checkConfiguration()) {
+            return;
+        }
         $id = json_decode($this->GetBuffer('CallID'));
         if ($id !== null) {
             VOIP_Disconnect($this->ReadPropertyInteger('VoIPInstanceID'), $id);
         }
     }
+
+    private function calculateInstanceStatus()
+    {
+        $getCode = function ()
+        {
+            if (($this->ReadPropertyInteger('VoIPInstanceID') === 0) || ($this->ReadPropertyInteger('TTSInstanceID') === 0)) {
+                return IS_INACTIVE;
+            }
+            $pollyConfiguration = json_decode(IPS_GetConfiguration($this->ReadPropertyInteger('TTSInstanceID')), true);
+            if (($pollyConfiguration['SampleRate'] !== '8000') || ($pollyConfiguration['OutputFormat'] !== 'pcm')) {
+                return IS_EBASE;
+            }
+            return IS_ACTIVE;
+        };
+
+        return $getCode();
```

Comment:
Die anonyme Funktion $getCode ist sinnlos, das kann direkt in calculateInstanceStatus